velox/exec/HashProbe.h

```diff
@@ -30,13 +31,45 @@ class HashProbe : public Operator {
       DriverCtx* driverCtx,
       const std::shared_ptr<const core::HashJoinNode>& hashJoinNode);
 
+  /// Define the internal execution state for hash build. The valid state
+  /// transition is depicted as follows:
+  ///
+  ///                           +--------------------------------+
+  ///                           ^                                |
+  ///                           |                                V
+  ///   kWaitForBuild -->  kRunning  -->  kWaitForPeers --> kFinish
+  ///                          ^               |
+  ///                          |               v
+  ///                          +---------------+
+  ///
+  enum class State {
```

Comment:
nit: Probably should be `private`?

Reply:
Yeah, but need to overload << operator for this class. Can we overload it for a private class?

Reply:
It's probably fine to leave it as public then